### PR TITLE
Chase's WDI radio

### PR DESCRIPTION
**Pull request by cpgruber**
I worked with Davis, it was good.

Completeness: 4 - i think we met all minimum requirements
Comfort level: 4 - starting to make better sense of Angular

### CODE REVIEW

**Comment by Mahartney** [comment]
Completeness: 5
Comfort: 3

Chase and I paired on this. I navigated most of the week and offered advice where I could. It went well and I learned a lot from Chase.

File: js/songs/form.directive.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+      templateUrl:"js/songs/_song_form.html",
+      replace:true,
+      scope:{
+        song:"=",
+        formType:"@"
+      },
+      link:function(scope){
+        scope.create = function(){
+          scope.song.$save(scope.song,function(song){
+            $state.go("songIndex",{},{reload:true})
```
Comment (jshawl) [inline js/songs/form.directive.js]: do you need the second two arguments here?

File: js/songs/form.directive.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        scope.delete = function(){
+          scope.song.$delete({id: scope.song.id},function(song){
+            $state.go("songIndex",{},{reload:true})
+          })
+        }
+        scope.update = function(){
+          scope.song.$update({id: scope.song.id})
+        }
+        scope.toggleForm = function(){
+          scope.formVisible = (scope.formVisible)?false:true;
```
Comment (jshawl) [inline js/songs/form.directive.js]: :+1: no need for parens here

File: js/songs/index.html [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<div class='song-form' data-form-directive data-song="SongIndexViewModel.newSong" data-form-type="new"></div>
+<div class='song-player' data-player-directive data-ng-show="SongIndexViewModel.playSong" data-song="SongIndexViewModel.playSong"></div>
+<div data-ng-repeat="song in SongIndexViewModel.songs | orderBy:'-created_at'">
```
Comment (jshawl) [inline js/songs/index.html]: :+1: nice use of filter

File: js/songs/show.directive.js [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      scope:{
+        song: "="
+      },
+      link: function(scope,el){
+        scope.song = SongFactory.get({id: scope.song.id});
+        scope.showBody = false;
+        scope.arrow = '\u2193';
+        scope.toggleShow = function(){
+          scope.showBody=(scope.showBody)?false:true;
+          scope.arrow=(scope.showBody)?"\u2191":"\u2193";
```
Comment (jshawl) [inline js/songs/show.directive.js]: awesome!

**Comment by jshawl** [comment]
:+1: excellent job!

**Pull request closed without merging**